docs/concepts/package-types.md

```diff
@@ -0,0 +1,225 @@
+
+```
+<modular-root>/dist/<your-package-name>
+├── dist-cjs
+|  ├── index.js
+|  ├── index.js.map
+|  ├── index2.js
+|  └── index2.js.map
+├── dist-es
+|  ├── index.js
+|  ├── index.js.map
+|  ├── index2.js
+|  └── index2.js.map
+├── dist-types
+|  └── index.d.ts
+└── package.json
+```
+
+Modular transpiles the `package` starting from its entrypoint twice: once with a
+target format of [CommonJS](https://nodejs.org/api/modules.html) in the
+`dist-cjs` directory and once with a target format of
+[ES Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules),
+in the `dist-es` directory. The output `package.json` links both compiled
+entrypoints respectively in the
+[`main`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#main) and
+[`module`](https://github.com/dherman/defense-of-dot-js/blob/master/proposal.md)
+field.
+
+Modular also extracts types from the source and outputs them in the `dist-types`
+directory, linking them in the
+[`typings`](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)
+manifest field. Here's an example of an output `package.json` generated when
+building a Modular `package`:
+
+```json
+{
+  "name": "<your-package-name>",
+  "private": false,
+  "modular": {
+    "type": "package"
+  },
+  "main": "dist-cjs/index.js",
+  "version": "1.0.0",
+  "module": "dist-es/index.js",
+  "typings": "dist-types/index.d.ts",
+  "dependencies": {},
+  "files": ["dist-cjs", "dist-es", "dist-types", "README.md"]
+}
+```
+
+### When to use the "package" type
+
+Modular `package`s are meant to provide re-usable functionality that can be
+published to third-party registries (like
+[the npm registry](https://www.npmjs.com/)) out-of-the-box; to underline this,
+Modular will, for example, refuse to work with packages that have the
+[`private`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#private)
+field set. You should use packages _only when there is a need to consume their
+build output_; if you just need to share some common source code inside your
+monorepo but you don't need to publish it externally, you should use the
+`source` type instead.
+
+## View
+
+Modular `view`s are Modular `package`s that, by convention, export a default
+React component. They are [created](../commands/add.md) with a default
+entrypoint is `src/index.tsx` in their manifest and they are built and tested
+exactly like `package` types. The only difference is that, since `view`s should
+always export a React component, they can be [`start`ed](../commands/start.md)
+to spawn a local developer server and render their default export to the DOM.
+
+Packages are generated by `modular add` using the
+[`modular-template-view`](https://github.com/jpmorganchase/modular/tree/main/packages/modular-template-view)
+[template](../concepts/templates.md)
+
+## Source
+
+Modular `source` types are packages that contain only source code in their `src`
```

Comment:
Don't `package`s also only contain source code in their `src` directory? I think the defining feature of the `source` type is: Modular `source` packages contain common source code that can be imported from other packages in the monorepo but cannot be built themselves.

Reply:
The subject of the adversative seems to refer to the "that" in the relative clause and sound a bit confusing. What do you think of:

> Modular `source` packages contain common source code that can be imported from
other packages in the monorepo. Source packages are not meant to be built
themselves; they are meant to be imported directly from other packages in the
monorepo.